examples/ldap/main.go

```
@@ -1,72 +1,43 @@
 package main
 
 import (
-	"crypto/tls"
-	"crypto/x509"
 	"flag"
-	"io/ioutil"
 	"net"
 	"net/http"
 	"os"
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/transport"
 	httptransport "github.com/go-kit/kit/transport/http"
-	"github.com/pkg/errors"
-	"github.com/spf13/afero"
-	"gopkg.in/yaml.v1"
 
 	"github.com/maverics-org/maverics/pkg/authn"
-	idp "github.com/maverics-org/maverics/pkg/connector/ldap"
-	idpmodel "github.com/maverics-org/maverics/pkg/model/idp"
+	"github.com/maverics-org/maverics/pkg/config"
+	"github.com/maverics-org/maverics/pkg/connector/ldap"
 )
 
-type Config struct {
-	LdapConfig idpmodel.LdapIDP
-	Address    string
-	TLS        TLS
-}
-
-type TLS struct {
-	CA   string
-	Cert string
-	Key  string
-}
-
 func main() {
-	logger := log.NewLogfmtLogger(os.Stderr)
-	configFile := flag.String("config", "authn.yaml", "Config file")
-
+	var configFilename string
+	flag.StringVar(&configFilename, "config", "authn.yaml", "Config file")
 	flag.Parse()
 
-	c, err := ioutil.ReadFile(*configFile)
-
-	if err != nil {
-		logger.Log("error", "could not read config file "+*configFile)
-		os.Exit(1)
-	}
+	logger := log.NewLogfmtLogger(os.Stderr)
 
-	var config Config
-	err = yaml.Unmarshal(c, &config)
+	configManager := config.NewConfigManager(logger)
+	conf, err := configManager.ReadConfig(configFilename)
 	if err != nil {
-		logger.Log("error", "could not parse config file "+*configFile)
+		_ = logger.Log("error", "failed to read config file")
```

Nit, here and throughout: `_ = logger.Log("level", "error, "msg", "...")`

---

Ongoing conversation in Slack, I'll wait until that's resolved.

Per Slack - I prefer explicitly ignoring errors, but I can see why it's very ugly when we have to do it with _every_ log statement. I'm open to ignoring that rule for our logger if it makes the code more readable.